modules/home-environment.nix

```diff
@@ -124,19 +126,21 @@ in
               '';
             };
 
-            mode = mkOption {
-              type = types.str;
-              default = "444";
-              description = "The permissions to apply to the file.";
+            executable = mkOption {
+              type = types.bool;
+              default = false;
+              description = "Make file executable.";
             };
           };
 
           config = {
             target = mkDefault name;
-            source = mkIf (config.text != null) (
-              let name' = "user-etc-" + baseNameOf name;
-              in mkDefault (pkgs.writeText name' config.text)
-            );
+            source = mkIf (config.text != null)
+                       (mkDefault (pkgs.writeTextFile {
+                          name = "home-file";
```

Comment:
I'm not so fond of this. I quite like to see sensible names when building the generation because it makes it easy to directly look at the right file in the Nix store and to look for other versions of the same file using shell globbing.

I.e.,
```
…
building path(s) ‘/nix/store/y96s9py4cs9gwxiqw05pn2i48ina0d7y-home-file-.Xresources’
building path(s) ‘/nix/store/pd9ppmnmqmf62vi51dmlmscblbmsbgbx-home-file-.bash_profile’
building path(s) ‘/nix/store/hlkfi82l7hbz8lxkdvpssjj92vb2i401-home-file-.bashrc’
building path(s) ‘/nix/store/3zhxv7f4vn3nrj3kqx0dc8fxlzdl1c2h-home-file-.gitconfig’
…
```
Vs.
```
…
building path(s) ‘/nix/store/x39p9a5y9prlxgkd51c5a61fmawzaw7x-home-file’
building path(s) ‘/nix/store/qzl0gig2s7iz94adma51p0hzhqy9m1xz-home-file’
building path(s) ‘/nix/store/kpyr73aai3bizrknkcmgnm7p2s1z0r7q-home-file’
building path(s) ‘/nix/store/jdqmnwd88pxmhi80lkfkf1plvw18aqmv-home-file’
…
```

So I would suggest changing this to
```
name = "home-file-" + safeBaseNameOf config.target;
```
where
```
safeBaseNameOf = f: replaceStrings [ " " ] [ "_" ] (baseNameOf f);
```
(or something similar).